File: 不靠谱的设计历史/T01.assets/_qa.js
const { chromium } = require('/Users/tao/Desktop/personal/uichat-mira/node_modules/.pnpm/playwright@1.60.0/node_modules/playwright');

(async () => {
  const browser = await chromium.launch({ executablePath: '/Users/tao/Library/Caches/ms-playwright/chromium_headless_shell-1234/chrome-headless-shell-mac-arm64/chrome-headless-shell' });
  const ctx = await browser.newContext({ viewport: { width: 375, height: 812 }, deviceScaleFactor: 2 });
  const page = await ctx.newPage();
  page.on('console', m => console.log(`[${m.type()}] ${m.text()}`));
  page.on('pageerror', e => console.log(`[error] ${e.message}`));

  const url = 'file://' + require('path').resolve('/Users/tao/Desktop/workspace/cardsUI/dist/T01.html');
  await page.goto(url, { waitUntil: 'networkidle' });
  await page.waitForTimeout(500);

  // 1) 基础态
  await page.screenshot({ path: '/Users/tao/Desktop/workspace/cardsUI/dist/T01.assets/qa-t01-375x812.png' });
  // 2) full page
  await page.screenshot({ path: '/Users/tao/Desktop/workspace/cardsUI/dist/T01.assets/qa-t01-fullpage.png', fullPage: true });

  // 3) 切换性别
  await page.click('.pf-radio[data-value="女"]');
  await page.waitForTimeout(150);
  await page.screenshot({ path: '/Users/tao/Desktop/workspace/cardsUI/dist/T01.assets/qa-t01-radio-female.png' });

  // 4) 填昵称
  await page.fill('#pf-nick', '诗得丽小芽');
  await page.waitForTimeout(150);

  // 5) 触发 pin 输入(点方块 → focus 容器 → keyboard.type)
  await page.click('#pf-pin');
  await page.waitForTimeout(100);
  await page.keyboard.type('123', { delay: 80 });
  await page.waitForTimeout(200);
  await page.screenshot({ path: '/Users/tao/Desktop/workspace/cardsUI/dist/T01.assets/qa-t01-pin-123.png' });

  // 6) 全部填写 + submit
  await page.keyboard.type('456', { delay: 80 });
  await page.waitForTimeout(200);
  await page.screenshot({ path: '/Users/tao/Desktop/workspace/cardsUI/dist/T01.assets/qa-t01-filled.png' });

  await browser.close();
  console.log('DONE');
})();

File: 不靠谱的设计历史/T01.assets/_qa.py
import asyncio
from playwright.async_api import async_playwright
import os, sys

async def main():
    html_path = os.path.abspath("/Users/tao/Desktop/workspace/cardsUI/dist/T01.html")
    url = "file://" + html_path
    out_dir = "/Users/tao/Desktop/workspace/cardsUI/dist/T01.assets"
    os.makedirs(out_dir, exist_ok=True)

    async with async_playwright() as p:
        browser = await p.chromium.launch()
        ctx = await browser.new_context(
            viewport={"width": 375, "height": 812},
            device_scale_factor=2,
        )
        page = await ctx.new_page()
        await page.goto(url, wait_until="networkidle")
        await page.wait_for_timeout(500)
        # 1) 375x812 视口截图(基础态)
        await page.screenshot(path=os.path.join(out_dir, "qa-t01-375x812.png"))
        # 2) full_page 看是否有滚动
        await page.screenshot(path=os.path.join(out_dir, "qa-t01-fullpage.png"), full_page=True)
        # 3) 调试:打印 console 错误
        page.on("console", lambda m: print(f"[{m.type}] {m.text}"))
        # 4) 模拟"女"被点击,确认 radio 视觉切换
        await page.click('.pf-radio[data-value="女"]')
        await page.wait_for_timeout(200)
        await page.screenshot(path=os.path.join(out_dir, "qa-t01-radio-female.png"))
        # 5) 在 pin 输入 123
        await page.click('.pf-pin')
        await page.keyboard.type("123", delay=80)
        await page.wait_for_timeout(200)
        await page.screenshot(path=os.path.join(out_dir, "qa-t01-pin-123.png"))
        # 6) 填昵称
        await page.fill("#pf-nick", "诗得丽小芽🌱")
        await page.wait_for_timeout(200)
        await page.screenshot(path=os.path.join(out_dir, "qa-t01-filled.png"))
        await browser.close()
        print("DONE")

asyncio.run(main())
